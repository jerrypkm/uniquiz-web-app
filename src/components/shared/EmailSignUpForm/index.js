import React from "react";

import Button from "@material-ui/core/Button";
import FormControl from "@material-ui/core/FormControl";
import FormHelperText from "@material-ui/core/FormHelperText";
import Input from "@material-ui/core/Input";
import InputLabel from "@material-ui/core/InputLabel";
import PropTypes from "prop-types";
import withStyles from "@material-ui/core/styles/withStyles";
import styles from "./styles";
import SignInLink from "../SignInLink";
import MoTypography from "components/library/MoTypography";
import { SIGN_UP } from "constants/i18n";
import Paper from "@material-ui/core/Paper";

const EmailSignUpForm = ({ classes, onSubmit, isInvalid, onChange, error }) => {
  return (
    <Paper className={classes.card}>
      <main className={classes.main}>
        <div className={classes.paper}>
          <MoTypography font="breeSerif" marginBottom="md" variant="h4">
            {SIGN_UP.CORE}
          </MoTypography>
          <form className={classes.form} onSubmit={onSubmit}>
            <FormControl margin="normal" required fullWidth>
              <InputLabel htmlFor="username">Name</InputLabel>
              <Input
                id="username"
                name="username"
                type="text"
                autoFocus
                onChange={onChange}
                placeholder="Mo Sharif"
              />
            </FormControl>
            <FormControl margin="normal" required fullWidth>
              <InputLabel htmlFor="email">Email</InputLabel>
              <Input
                id="email"
                name="email"
                type="email"
                autoFocus
                autoComplete="email"
                onChange={onChange}
                placeholder="me@moskool.com"
              />
            </FormControl>
            <FormControl margin="normal" required fullWidth>
              <InputLabel htmlFor="passwordOne">Password</InputLabel>
              <Input
                id="passwordOne"
                name="passwordOne"
                type="password"
                onChange={onChange}
                placeholder="Password"
              />
            </FormControl>
            <FormControl margin="normal" required fullWidth>
              <InputLabel htmlFor="passwordTwo">Confirm Password</InputLabel>
              <Input
                id="passwordTwo"
                name="passwordTwo"
                type="password"
                onChange={onChange}
                placeholder="Password"
              />
            </FormControl>
            <Button
              type="submit"
              fullWidth
              variant="contained"
              color="primary"
              className={classes.submit}
              disabled={isInvalid}
            >
              Sign Up
            </Button>
            {error && (
              <FormHelperText
                className={classes.error}
                id="component-error-text"
                margin="dense"
                error
                focused
              >
                {error.message}
              </FormHelperText>
            )}
          </form>
          <SignInLink />
        </div>
      </main>
    </Paper>
  );
};

EmailSignUpForm.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(EmailSignUpForm);
