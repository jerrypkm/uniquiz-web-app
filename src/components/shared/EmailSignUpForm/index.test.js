import React from "react";

import ShallowRenderer from "react-test-renderer/shallow";
import EmailSignUpForm from "./index.js";

const renderer = new ShallowRenderer();

describe("EmailSignUpForm Component", () => {
  test("should match the snapshot", () => {
    renderer.render(<EmailSignUpForm />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
