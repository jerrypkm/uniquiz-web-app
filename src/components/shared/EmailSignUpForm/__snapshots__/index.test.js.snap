// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailSignUpForm Component should match the snapshot 1`] = `
<EmailSignUpForm
  classes={
    Object {
      "card": "EmailSignUpForm-card-2",
      "error": "EmailSignUpForm-error-6",
      "form": "EmailSignUpForm-form-4",
      "main": "EmailSignUpForm-main-1",
      "paper": "EmailSignUpForm-paper-3",
      "submit": "EmailSignUpForm-submit-5",
    }
  }
/>
`;
